extensions/odoo_theme/static/js/utils.js

```diff
@@ -4,63 +4,72 @@ let tocEntryListId = 0; // Used to generate IDs of toc entry lists for both the
+ *                <a href="#heading_without_child">text</a>
+ *            </li>
+ *            <li>
+ *                <a href="#heading_with_children" class="o_toc_entry_wrapper">
+ *                    <i class="i-chevron-right" data-bs-target="#o_target_{id}" data-bs-toggle="collapse"/>
+ *                    <span>text</span>
+ *                </a>
+ *                <ul id="o_target_{id}" class="collapse">...</ul>
+ *            </li>
+ *        </ul>
+ *    </li>
+ *</ul>
  *
- * @param {HTMLElement} tocElement - The element containing the TOC
+ * @param {HTMLElement} tocElement - The element containing the TOC.
  */
 const _prepareAccordion = (tocElement) => {
-    // Start at the second TOC entry list (<ul>) to avoid collapsing the entire TOC
+    // Start at the second TOC entry list (<ul>) to avoid collapsing the entire TOC.
     const tocRoot = tocElement.querySelector('ul');
     tocRoot.querySelectorAll('ul').forEach(tocEntryList => {
-        // Modify the <ul> element
+        // Modify the <ul> element.
         tocEntryList.id = `o_target_${tocEntryListId++}`;
         tocEntryList.classList.add('collapse');
-        // Create and configure an <i> element
-        const arrowButton = document.createElement('I');
-        arrowButton.setAttribute('data-bs-target', `#${tocEntryList.id}`);
-        arrowButton.setAttribute('data-bs-toggle', 'collapse');
-        arrowButton.classList.add('i-chevron-right');
-        // Modify the <a> element (only if it has no href, otherwise let the redirection happen)
+        // Modify the <a> element (only if it has no href, otherwise let the redirection happen).
         const relatedHeadingRef = tocEntryList.previousSibling;
+        relatedHeadingRef.classList.add('o_toc_entry_wrapper');
         if (relatedHeadingRef.getAttribute('href') === '#') {
             relatedHeadingRef.setAttribute('data-bs-target', `#${tocEntryList.id}`);
             relatedHeadingRef.setAttribute('data-bs-toggle', 'collapse');
         }
-        // Create a <div> element
-        const tocEntryWrapper = document.createElement('DIV');
-        tocEntryWrapper.classList.add('o_toc_entry_wrapper');
-        // Insert the <i> and <a> elements inside the <div> and prepend the <div> to the <ul>
-        tocEntryWrapper.append(arrowButton, relatedHeadingRef);
-        tocEntryList.parentNode.insertBefore(tocEntryWrapper, tocEntryList);
+        // Create and configure an <i> element.
+        const arrowButton = document.createElement('I');
+        arrowButton.setAttribute('data-bs-target', `#${tocEntryList.id}`);
+        arrowButton.setAttribute('data-bs-toggle', 'collapse');
+        arrowButton.classList.add('i-chevron-right');
+        // Create a <span> element with the text of the <a>.
+        const span = document.createElement('SPAN');
+        span.innerText = relatedHeadingRef.innerText
+        relatedHeadingRef.innerText = null;
+        // Insert the <i> and <span> elements inside the <a>.
```

Comment:
This brings a regression: it does not longer just expand the accordion when you click the arrow of "Tutorials" from the index; instead, the animation is aborted and you are redirected to /tutorials.html.